### PR TITLE
Record tests for national cloud better

### PR DESCRIPTION
**Pull request by matthchr**
Also add support for billing invoice ID test variable

### CODE REVIEW

File: v2/internal/config/vars_test.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 	"reflect"
 	"testing"
 
+	// TODO: Importing this for side effects is required as it initializes cloud.AzurePublic
```
Comment (theunrepentantgeek) [inline v2/internal/config/vars_test.go:10]: Don't think this needs `TODO`.
```suggestion
	// Importing this for side effects is required as it initializes cloud.AzurePublic
```

File: v2/internal/testcommon/test_context.go [hunk 2]
```diff
@@ -93,28 +96,31 @@ func (tc TestContext) ForTest(t *testing.T, cfg config.Values) (PerTestContext,
 	logger := NewTestLogger(t)
 
 	cassetteName := "recordings/" + t.Name()
-	creds, azureIDs, recorder, err := createRecorder(cassetteName, tc.RecordReplay)
+	details, err := createRecorder(cassetteName, cfg, tc.RecordReplay)
 	if err != nil {
 		return PerTestContext{}, errors.Wrapf(err, "creating recorder")
 	}
+	// Use the recorder-specific CFG, which will URLs and AADAuthorityHost (among other things) to default
```
Comment (theunrepentantgeek) [inline v2/internal/testcommon/test_context.go:103]: Missing word?

```suggestion
	// Use the recorder-specific CFG, which will force URLs and AADAuthorityHost (among other things) to default
```

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/internal/testcommon/test_context.go (hunk 2 above)
Comment (matthchr) [reply v2/internal/testcommon/test_context.go:103]: Thanks, fixed

File: v2/internal/config/vars_test.go (hunk 1 above)
Comment (matthchr) [reply v2/internal/config/vars_test.go:10]: Good point

**Review by super-harsh — APPROVED** [review]

**Pull request merged**